libs/utils/src/JobSystem.cpp

```diff
@@ -230,9 +223,9 @@ void JobSystem::wake() noexcept {
 }
 
 inline JobSystem::ThreadState& JobSystem::getState() noexcept {
-    // check we're not using a thread not owned by the thread pool
-    assert(sThreadState);
-    return *sThreadState;
+    auto iter = mThreadMap.find(std::this_thread::get_id());
+    ASSERT_PRECONDITION(iter != mThreadMap.end(), "This thread has not been adopted.");
```

Comment:
I replaced `assert` with `ASSERT_PRECONDITION` because this was difficult to debug in release config.